common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -397,48 +381,54 @@ public static boolean isIpV6AddressesPreferred() {
         return null;
     }
 
+    /**
+     * Convert ASCII hexadecimal character to the {@code int} value.
+     * Unlike {@link Character#digit(char, int)}, returns {@code 0} if character is not a HEX-represented.
+     */
     private static int getIntValue(char c) {
-        switch (c) {
-            case '0':
-                return 0;
-            case '1':
-                return 1;
-            case '2':
-                return 2;
-            case '3':
-                return 3;
-            case '4':
-                return 4;
-            case '5':
-                return 5;
-            case '6':
-                return 6;
-            case '7':
-                return 7;
-            case '8':
-                return 8;
-            case '9':
-                return 9;
+        if (c >= '0' && c <= '9') {
+            return c - '0';
         }
-
-        c = Character.toLowerCase(c);
-        switch (c) {
-            case 'a':
-                return 10;
-            case 'b':
-                return 11;
-            case 'c':
-                return 12;
-            case 'd':
-                return 13;
-            case 'e':
-                return 14;
-            case 'f':
-                return 15;
+        if (c >= 'A' && c <= 'F') {
+            // 0xA - a start value in sequence 'A'..'F'
+            return c - 'A' + 0xA;
+        }
+        if (c >= 'a' && c <= 'f') {
+            // 0xA - a start value in sequence 'a'..'f'
+            return c - 'a' + 0xA;
         }
         return 0;
     }
 
+    private static int decimalDigit(String str, int pos) {
+        return str.charAt(pos) - '0';
+    }
+
+    private static byte ipv4WordToByte(String ip, int from, int toExclusive) {
+        int ret = decimalDigit(ip, from);
```

Comment:
Checked: benchmarks shows performans degradation for the variant with `switch-case`:

```java
Benchmark                                       (ip)   Mode  Cnt    Score   Error   Units
// current
IpV4ToBytesBenchmark.ipv4ToBytesNew        127.0.0.1  thrpt    6   75,815 ± 0,405  ops/us
IpV4ToBytesBenchmark.ipv4ToBytesNew  255.255.255.255  thrpt    6   49,994 ± 3,282  ops/us
IpV4ToBytesBenchmark.ipv4ToBytesNew          1.1.1.1  thrpt    6  113,590 ± 5,029  ops/us

// with switch-case
IpV4ToBytesBenchmark.ipv4ToBytesNew2       127.0.0.1  thrpt    6   61,477 ± 5,579  ops/us
IpV4ToBytesBenchmark.ipv4ToBytesNew2 255.255.255.255  thrpt    6   42,536 ± 0,437  ops/us
IpV4ToBytesBenchmark.ipv4ToBytesNew2         1.1.1.1  thrpt    6   67,125 ± 2,029  ops/us
```